source/common/network/utility.cc

```diff
@@ -355,8 +355,19 @@ Address::InstanceConstSharedPtr Utility::getOriginalDst(Socket& sock) {
         sock.getSocketOption(SOL_IPV6, IP6T_SO_ORIGINAL_DST, &orig_addr, &addr_len).return_value_;
   }
 
+#ifdef IP_TRANSPARENT
+  socklen_t flag_len = sizeof(int);
+  int is_transparent;
+  if (status != 0) {
+    status = sock.getSocketOption(SOL_IP, IP_TRANSPARENT, &is_transparent, &flag_len).return_value_;
+    if (status != 0 || !is_transparent) {
+      return nullptr;
+    }
+    safeMemcpy(reinterpret_cast<sockaddr*>(&orig_addr), sock.ioHandle().localAddress()->sockAddr());
+#else
   if (status != 0) {
     return nullptr;
+#endif
```

Comment:
if the IP transparent doesn't support on the os, and socket is transparent, should we log a warning here.

Reply:
Dont know why socket could be transparent when os doesnt support IP_TRANSPARENT...

Reply:
Haha, right, sorry for this stupid comment. Please ignore it.